### PR TITLE
Express Application for Json Validation

### PR DESCRIPTION
**Pull request by nadine-piveteau**
Small express web application to validate json-files based on `validat.js`.

Test: 
`node index.js`
Application runs on localhost:9010. You can test it by means of a post request with `geojsons/ch.meteoschweiz.messdaten-niederschlag.json` in the body.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+app.use(compression());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json({limit: '50mb'}));
+
+
+function testHeader(headers){
+  var testCache = (headers['cache-control'] == 'no-cache');
+  var contentType = (headers['content-type'] == 'application/json');
+  var encoding = headers['accept-encoding'].split(", ").includes('gzip');
+  return testCache && contentType && encoding
```
Comment (loicgasser) [inline index.js]: `;` missing at the end

File: index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+app.use(compression());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json({limit: '50mb'}));
+
+
+function testHeader(headers){
+  var testCache = (headers['cache-control'] == 'no-cache');
+  var contentType = (headers['content-type'] == 'application/json');
+  var encoding = headers['accept-encoding'].split(", ").includes('gzip');
```
Comment (loicgasser) [inline index.js]: all these headers could not exist
check that the key is defined in the headers first

File: index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+// send validated json to s3
+app.post('/push', function(req, res) {
+  var jsonObject = req.body;
+  if(validator.isValid(jsonObject) && testHeader(req.headers)){
+   //set params for S3 entry
+   var uploadParams = {Bucket: '', Key: '', Body: ''};
+   uploadParams.Bucket = process.env.BUCKET_NAME;
+   uploadParams.Key = jsonObject["mapname"] + '.json';
+   console.log("Key of S3 object: " + uploadParams.Key);
```
Comment (loicgasser) [inline index.js]: Ok for now but we want to use a logger to filter what show according to the env.
Maybe
https://github.com/bithavoc/express-winston

File: index.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+app.get('/start', function (req, res) {
+  res.send('Start successfull!');
+});
+
+
+// send validated json to s3
+app.post('/push', function(req, res) {
+  var jsonObject = req.body;
+  if(validator.isValid(jsonObject) && testHeader(req.headers)){
+   //set params for S3 entry
```
Comment (loicgasser) [inline index.js]: 2 spaces

File: index.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+   s3.upload (uploadParams, function (err, data) {
+    if (err) {
+      console.log("Error", err);
+    }
+    if (data) {
+      console.log("Upload Success", data.Location);
+    }
+   });
+  }else{
+    console.log('Json is not valid.');
```
Comment (loicgasser) [inline index.js]: please return the error of the validator instead. Can be done later.

File: validate.js [hunk 6]
```diff
@@ -28,8 +31,28 @@ var featurePropertiesSchema = {
...
-var valid = validate(c)
-if (!valid) console.log(validate.errors);
+
+
+function getBBox(epsg){
+  var xmin = epsgBBox[epsg][0];
+  var ymin = epsgBBox[epsg][1];
+  var xmax = epsgBBox[epsg][2];
+  var ymax = epsgBBox[epsg][3];
+  return [xmin, ymin, xmax, ymax];
```
Comment (loicgasser) [inline validate.js]: lol, why not simply `return epsgBBox[epsg];`

Comment (nadine-piveteau) [reply validate.js]: haha of course, stupid me xD

**Comment by loicgasser** [comment]
Thanks!
Now we need to focus on error reporting, would be good to know which line is failing.

**Pull request merged**